style-sync/src/main/scala/commercialtools/stylesync/StyleSync.scala

```diff
@@ -0,0 +1,146 @@
+package commercialtools.stylesync
+
+import java.io.File
+import java.nio.file.Paths
+
+import com.google.api.ads.dfp.axis.v201708.{CreativeTemplate, CreativeTemplateType, NativeStyle}
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.io.Source
+
+trait Logging {
+  val logger: Logger = LoggerFactory.getLogger(getClass)
+}
+
+case class GuStyle(formatName: String,
+                   dfpFormat: Option[CreativeTemplate],
+                   styleName: String,
+                   dfpStyle: Option[NativeStyle],
+                   localPath: String,
+                   testFile: Option[File]) {
+
+  val localHtmlLines: List[String] = Source.fromFile(Paths.get(localPath, "index.html").toFile).getLines.toList
+  val localCssLines: List[String] = Source.fromFile(Paths.get(localPath, "index.css").toFile).getLines.toList
+}
+
+object StyleSync extends App with Logging {
+
+  /*
+     Builds up a list of GuStyle objects, which link locally built templates to their
+     respective formats (CreativeTemplates) and styles (NativeStyles) in DFP.
+  */
+  def findStyles(buildRootDirectory: File,
+                 nativeFormats: => List[CreativeTemplate],
+                 nativeStyles: => List[NativeStyle]): List[GuStyle] = {
+
+    def titleCase(str: String, delim: String = " "): String =
+      str.split(delim).map(word =>
+        word.head.toUpper + word.tail.toLowerCase
+      ).mkString(delim)
+
+    def deriveFormatName(formatDirectory: String) =
+      titleCase(formatDirectory.replaceAll("-", " "))
+      .replaceAll("Capi", "cAPI")
+
+    def deriveStyleName(styleDirectory: String, formatName: String) = {
+      val platform = if(styleDirectory == "amp") "AMP" else styleDirectory.capitalize
```

Comment:
Is `AMP` the only directory that needs to be uppercase? Are all the rest of the form `Foo`?

Reply:
Yep, I'm sort of deriving convention from what is already set up in DFP.